### ISSUE TITLE
🌟Feature request: Popover with modal option

### ISSUE THREAD

**Issue by xRSquared** [melt-ui/melt-ui#526]
**Is your feature request related to a problem? Please describe.**

I want to have the popover block all user interaction outside of the popover. Essentially, I want the option for the popover to be a modal.

**Describe the solution you'd like**

Allow for a modal option in `createPopover()` like the option in Radix, or add an option to disable outside pointer events.

**Describe alternatives you've considered**
I can implement this outside of melt-ui, but I feel like this could be integrated into melt-ui, and I suspect it is a common use case.

**Comment by wesharper** [comment]
Would the dialog builder be inappropriate for your use-case?

**Comment by xRSquared** [comment]
I'm actually looking for a hybrid between a Dialog and a Popover. I'd like the dialog content to be positioned relative to the trigger button, similar to a Popover, while also preventing pointer events with elements outside of it, like a Dialog. Achieving this would involve combining elements from both builders.

It would be great to have an `asModal` option similar to what Radix offers. With this option, the content could receive the dialog role, and we could control its positioning using Popper and passing the `positioning` to `createPopover.` It would look something like this:

```ts
	const {
		elements: { trigger, content, arrow, close },
		states: { open }
	} = createPopover({
		positioning: {
			placement: 'top-end',
			gutter: 16
		},
                asModal: true
	});
```

Implementing this `asModal` option would greatly enhance flexibility and functionality, allowing users to efficiently create the desired hybrid behavior without the need for complex workarounds. I believe that incorporating this option would significantly improve the developer experience for this particular use case.

**Comment by wesharper** [comment]
Personally, I think this would be better achieved with CSS, perhaps it would be worth building something of a popover example in the dialog example docs.

**Comment by TGlide** [comment]
> Personally, I think this would be better achieved with CSS, perhaps it would be worth building something of a popover example in the dialog example 

I think it's a valid feature request, specially because of a11y reasons. At least at a first glance.

**Comment by huntabyte** [comment]
Closed by #1025
